src/main/java/edu/harvard/iq/dataverse/engine/command/impl/RegisterOIDCUserCommand.java

```diff
@@ -0,0 +1,129 @@
+        super(aRequest, (DvObject) null);
+        this.bearerToken = bearerToken;
+        this.userDTO = userDTO;
+    }
+
+    @Override
+    protected void executeImpl(CommandContext ctxt) throws CommandException {
+        try {
+            OIDCUserInfo oidcUserInfo = ctxt.authentication().verifyOIDCBearerTokenAndGetUserIdentifier(bearerToken);
+            UserRecordIdentifier userRecordIdentifier = oidcUserInfo.getUserRecordIdentifier();
+
+            if (ctxt.authentication().lookupUser(userRecordIdentifier) != null) {
+                throw new IllegalCommandException(BundleUtil.getStringFromBundle("registerOidcUserCommand.errors.userAlreadyRegisteredWithToken"), this);
+            }
+
+            UserInfo userClaimsInfo = oidcUserInfo.getUserClaimsInfo();
+            boolean provideMissingClaimsEnabled = FeatureFlags.API_BEARER_AUTH_PROVIDE_MISSING_CLAIMS.enabled();
+
+            updateUserDTO(userClaimsInfo, provideMissingClaimsEnabled);
+
+            AuthenticatedUserDisplayInfo userDisplayInfo = new AuthenticatedUserDisplayInfo(
+                    userDTO.getFirstName(),
+                    userDTO.getLastName(),
+                    userDTO.getEmailAddress(),
+                    userDTO.getAffiliation() != null ? userDTO.getAffiliation() : "",
+                    userDTO.getPosition() != null ? userDTO.getPosition() : ""
+            );
+
+            Map<String, String> fieldErrors = validateUserFields(ctxt, provideMissingClaimsEnabled);
+            if (!fieldErrors.isEmpty()) {
+                throw new InvalidFieldsCommandException(
+                        BundleUtil.getStringFromBundle("registerOidcUserCommand.errors.invalidFields"),
+                        this,
+                        fieldErrors
+                );
+            }
+
+            ctxt.authentication().createAuthenticatedUser(userRecordIdentifier, userDTO.getUsername(), userDisplayInfo, true);
+
+        } catch (AuthorizationException ex) {
+            throw new PermissionException(ex.getMessage(), this, null, null, true);
+        }
+    }
+
+    private void updateUserDTO(UserInfo userClaimsInfo, boolean provideMissingClaimsEnabled) {
+        if (provideMissingClaimsEnabled) {
+            // Update with available OIDC claims, keep existing values if claims are absent
+            userDTO.setUsername(getValueOrDefault(userClaimsInfo.getPreferredUsername(), userDTO.getUsername()));
+            userDTO.setFirstName(getValueOrDefault(userClaimsInfo.getGivenName(), userDTO.getFirstName()));
+            userDTO.setLastName(getValueOrDefault(userClaimsInfo.getFamilyName(), userDTO.getLastName()));
+            userDTO.setEmailAddress(getValueOrDefault(userClaimsInfo.getEmailAddress(), userDTO.getEmailAddress()));
+        } else {
+            // Always use the claims provided by the OIDC provider, regardless of whether they are null or not
+            userDTO.setUsername(userClaimsInfo.getPreferredUsername());
```

Comment:
Per the comment earlier - should we fail here if you send JSON with these values/with conflicting values?